shared/output/cloud/vision_v1/lib/google/cloud/vision/v1/image_annotator/client.rb

```diff
@@ -34,6 +35,8 @@ module V1
         module ImageAnnotator
           # Service that implements ImageAnnotator API.
           class Client
+            include Google::Cloud::Vision::V1::Paths
```

Comment:
As noted earlier, it so happens that all the path helpers apply to the ProductSearch client, but the global paths helper is now also included in ImageAnnotator for which the helpers are not relevant. This brings up a shortcoming of our strategy to make Paths a package thing rather than a service thing. So now I'm kind of torn about it. However, the alternative is to do an analysis to determine which messages/resources are relevant to which clients, which may be more work than it's worth.

So instead, I have a thought of not including Paths in the client classes by default. (Users can still do so themselves by running `Google::Cloud::Vision::V1::Paths.extend my_client` if they want.) That way we sidestep this issue, and also sidestep the possibility of collisions between client methods and path helpers (by making it the user's responsibility).

Reply:
> As noted earlier, it so happens that all the path helpers apply to the ProductSearch client, but the global paths helper is now also included in ImageAnnotator for which the helpers are not relevant. This brings up a shortcoming of our strategy to make Paths a package thing rather than a service thing.

I don’t see the shortcoming. Why is that a bad thing?

Reply:
The shortcoming is that there are now a bunch of methods (the path helpers) on the `ImageAnnotator::Client` class that, in the best case have no benefit because users will never use them with this client, and in the worst case can collide with endpoint methods.

Reply:
I would strongly prefer if the path helpers were included on the client. The devex is quite awful without it, and it leaks a lot of internal abstractions when it's not.

Reply:
An alternate approach is to walk the request objects looking for all fields that have a resource references and including only the associated resources.

Reply:
@blowmage That seems like a lot of work. And would it mean moving the Paths modules back under the service modules because now each service has paths for a different subset of the resources?

Reply:
Just different work. We could abandon this PR, then open an alternate PR that changes the way resources are identified. If you are concerned about having helper methods with no benefit then I see no other option than to inspect all the request objects for the service.

Reply:
We can try that. If it's tractable, I'd be a lot happier with having only the relevant path helpers on each client.